pkg/controller/binding/binding_controller.go

```diff
@@ -0,0 +1,215 @@
+
+	// IF Consolidated doesn't exists, let's create it.
+	if err != nil && errors.IsNotFound(err) {
+		// Getting the current consolidated object failed due to it being non-existent.
+		// Let's create it!
+		log.Printf("Reconcile: Creating new Consolidated object: %s/%s.", currentConsolidated.Namespace, currentConsolidated.Name)
+		consolidated, err := apiv1alpha1.NewConsolidatedFromBinding(binding, c)
+		if err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+		trueVar := true
+		consolidated.SetOwnerReferences(append(consolidated.GetOwnerReferences(), v1.OwnerReference{
+			APIVersion: "api.3scale.net/v1alpha1",
+			Kind:       "Binding",
+			Name:       binding.Name,
+			UID:        binding.UID,
+			Controller: &trueVar,
+		}))
+
+		err = c.Create(context.TODO(), consolidated)
+		if err != nil {
+			return reconcile.Result{Requeue: true}, err
+		}
+
+	} else if err != nil {
+		// Something is broken when trying to get the existing consolidated Object
+		log.Printf("error: %s", err)
+		return reconcile.Result{Requeue: true}, err
+	} else {
+		// Consolidated Object Already exists
+		// Let's compare those, first, we should calculate the desired Consolidated object.
+		desiredConsolidated, err := apiv1alpha1.NewConsolidatedFromBinding(binding, c)
+		if err != nil {
+			log.Printf("error: %s", err)
+			return reconcile.Result{Requeue: true}, err
+		}
+		// Compare with the current consolidated object.
+		if compareConsolidated(currentConsolidated, desiredConsolidated) {
+			// Desired and existing are equal, nothing to do.
+			log.Printf("Skip reconcile: Consolidated config %s/%s ok.", currentConsolidated.Namespace, currentConsolidated.Name)
+		} else {
+			// Consolidated Objects are different, let's update the existing one with the desired object.
+			trueVar := true
+			// We set the binding as the owner of the consolidated object, if one gets removed, this one would  too
+			desiredConsolidated.SetOwnerReferences(append(desiredConsolidated.GetOwnerReferences(), v1.OwnerReference{
+				APIVersion: "api.3scale.net/v1alpha1",
+				Kind:       "Binding",
+				Name:       binding.Name,
+				UID:        binding.UID,
+				Controller: &trueVar,
+			}))
+			// Set the proper Meta from the existing object.
+			desiredConsolidated.ObjectMeta = currentConsolidated.ObjectMeta
+			err := c.Update(context.TODO(), desiredConsolidated)
+			if err != nil {
+				// Something went wrong when trying to update the actual consolidated object.
+				log.Printf("error: %s", err)
+				return reconcile.Result{Requeue: true}, err
+			}
+			// All done here
+			log.Printf("Reconcile: Consolidated config %s/%s has been updated.", currentConsolidated.Namespace, currentConsolidated.Name)
+		}
+	}
+	// Return without errors and resume the reconcile loop
+	return reconcile.Result{}, nil
+}
+
+func compareConsolidated(consolidatedA, consolidatedB *apiv1alpha1.Consolidated) bool {
+	//Let's compare only the Spec
+	A, _ := json.Marshal(consolidatedA.Spec)
```

Comment:
why serialize to json byte stream? `reflect.DeepEqual` should compare structs deeply.

Reply:
does not apply. For our structs, two array of elements may be equal even when do not follow same order. Conceptually `Set` data structure should be used, unfortunately not available in Go.

Reply:
Yup! we need to write sort functions for those, check: https://github.com/3scale/api-operator/blob/4856b4d1f34fdf22f4202ffc1e4447166983b75f/pkg/apis/3scale/v1alpha1/types.go#L55